core/action/action.go

```diff
@@ -20,13 +25,21 @@ type Factory interface {
 
 // Runner runs actions
 type Runner interface {
-	//Run the specified Action
 	Run(context context.Context, action Action, uri string, options interface{}) (code int, data interface{}, err error)
+
+	//Run the specified Action
+	RunAction(context context.Context, actionID string, inputGenerator InputGenerator, options map[string]interface{}) (code int, data map[string]interface{}, err error)
+}
+
+//TODO REVIEW - Need a way to package the output of the trigger, its metadata, and the corresponding mapper
+//TODO        - so it doesn't get evaluated until Action.Run, probably needs a better name
+type InputGenerator interface {
```

Comment:
I am thinking InputsFactory would be a better name

Reply:
Not sure about "factory"...  let me think about..